dio/lib/src/file_access_mode.dart

```diff
@@ -0,0 +1,26 @@
+/// {@template dio.FileAccessMode}
+///  The file access mode when downloading the file.
+/// - [FileAccessMode.write]: Mode for opening a file for reading and writing.
+///    The file is overwritten if it already exists. The file is created if it
+///    does not already exist
+/// - [FileAccessMode.append]: Mode for opening a file for reading and writing
+///    to the end of it. The file is created if it does not already exist.
+/// {@endtemplate}
+enum FileAccessMode {
+  write,
+  append,
+}
```

Comment:
Can we put the enumeration in `src/options.dart` and split the document?

```suggestion
/// {@template dio.options.FileAccessMode}
/// The file access mode when downloading a file, corresponds to a subset of 
/// dart:io::[FileMode].
/// {@endtemplate}
enum FileAccessMode {
  /// Mode for opening a file for reading and writing. The file is overwritten
  /// if it already exists. The file is created if it does not already exist.
  write,

  /// Mode for opening a file for reading and writing to the end of it.
  /// The file is created if it does not already exist.
  append,
}
```